### PR TITLE
[Java][Client] Fix #12556 Support primitives and arrays in oneOf

### LINKED ISSUE #12556: openapi-generator-maven-plugin generates invalid Java code when using "oneOf:" property
#### Description

Maven test target in openapi-generator-maven-plugin will run successfully,
**but** the generated Java models are not valid Java code and cannot be compiled by Java compiler
and therefore cannot be used as a JavaClient.

Because of the following issues in the genrated Java model `Value.java`:
```
Array.validateJsonObject(jsonObject); // The method validateJsonObject(JsonObject) is undefined for the type Array
```
Note, the same error can be reproduced with the Java Types, `String, Boolean, BigDecimal`

#### Steps to reproduce

When using following modified schema in openapi\modules\openapi-generator-maven-plugin\src\test\resources\petstore-on-classpath.yaml
```
openapi: 3.0.0
servers:
  - url: 'http://petstore.swagger.io/v2'
info:
  description: >-
    This is a sample server Petstore server. For this sample, you can use the api key
    `special-key` to test the authorization
  version: 1.0.0
  title: OpenAPI Petstore
  license:
    name: Apache-2.0
    url: 'https://www.apache.org/licenses/LICENSE-2.0.html'
...
paths: 
  /values:  
    get:  
      tags:  
        - values  
      summary: Get some primitive values  
      description: ''  
      operationId: getSomeValues  
      responses:  
        '200':  
          description: successful operation  
          content:  
            application/json:  
              schema:  
                $ref: '#/components/schemas/Value'  
        '400':  
          description: Invalid Value  
...

schemas:  
    Value:  
      description: Value datatype  
      oneOf:  
        - $ref: "#/components/schemas/Scalar"  
        - $ref: "#/components/schemas/Array"  
    Scalar:  
      description: Scalar datatype  
      oneOf:  
        - type: string  
          maxLength: 1089  
        - type: number  
        - type: boolean  
    Array:  
      type: array  
      minItems: 1  
      items:  
        $ref: "#/components/schemas/Scalar"
```

And the following modified `openapi\modules\openapi-generator-maven-plugin\pom.xml`,  
the `<library>jersey2</library>` was removed.
```
 <configuration>
                    <inputSpec>petstore-on-classpath.yaml</inputSpec>
                    <generatorName>java</generatorName>

                    <skipIfSpecIsUnchanged>true</skipIfSpecIsUnchanged>

                    <configOptions>
                        <dateLibrary>joda</dateLibrary>
                    </configOptions>
                    <output>${basedir}/target/generated-sources/common-maven/remote-openapi</output>
                    <apiPackage>remote.org.openapitools.client.api</apiPackage>
                    <modelPackage>remote.org.openapitools.client.model</modelPackage>
                    <invokerPackage>remote.org.openapitools.client</invokerPackage>
</configuration>
```

##### openapi-generator version
Can be reproduced with the current "Master" version of this repository.
But also the compile error was reproduced on a build server with the openapi-generator-maven-plugin:
```
<!-- RELEASE_VERSION -->
<version>6.0.0</version>
<!-- /RELEASE_VERSION -->
```
##### Suggest a fix
Maybe there should be a more strict check in this code section,
https://github.com/OpenAPITools/openapi-generator/blob/69f79fb7892948590a9ffe46754c47ddd2634be1/modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/oneof_model.mustache#L101-L123

It looks like this would generate .validateJsonObject(jsonObject) on every type which is not valid
for type `Array, String, Boolean, BigDecimal` in Java.

### PR DESCRIPTION
**Pull request by karzang**
As described in #12556 it is not possible to use schemas with primitive types or schemas of array type since the java client generator does only support schemas of type 'object' in oneOf. This change supports the use of primitives and arrays in oneOf without the need to wrap them into objects.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2022/09) @lwlee2608 (2019/10)

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

### CODE REVIEW

**Comment by karzang** [comment]
<html>
<body>


@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)
--



</body>
</html>

**Comment by wing328** [comment]
@dinuta please bulid the JAR locally to help test this PR.

```
git checkout -b karzang-issue_12556 master
git pull https://github.com/karzang/openapi-generator.git issue_12556
mvn clean package -DskipTests -Dmaven.javadoc.skip=true
```

then run you tests with `modules/openapi-generator-cli/target/openapi-generator-cli.jar`

**Comment by dinuta** [comment]
Hi team,

Made 1 test with the following config (to generate a client consumer):  
{
...
  "library": "rest-assured",
  "serializationLibrary": "jackson"
...
}

The classes seems fine, but there aren't the swagger annotations on the api classes as per 6.2.1 version:
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiResponse;
import io.swagger.annotations.ApiResponses;

Example annotations:
@ApiModelProperty(value = "The name of the files to pick up.")
@Api(value = "SomeApiProfiles")

It might not be so relevant for the issue, but I'll try to generate for okhttp3 with gson also. Don't know when I'll finish.

Any questions ? Thoughts?

**Comment by karzang** [comment]
@dinuta @wing328  Has this PR a chance of being merged soon?

**Comment by wing328** [comment]
@karzang sorry for the delay. can you please PM me via Slack https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g to quickly walk through this PR?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenComposedSchemas.java [hunk 1]
```diff
@@ -47,6 +47,22 @@ public CodegenProperty getNot() {
...
+    }
+
+    public void setAnyOf(List<CodegenProperty> anyOf) {
+        this.anyOf = anyOf;
+    }
+
+    public void setNot(CodegenProperty not) {
+        this.not = not;
+    }
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenComposedSchemas.java:65]: FYI @OpenAPITools/generator-core-team

File: samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/model/BigCatAllOf.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
+/*
```
Comment (wing328) [inline samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/model/BigCatAllOf.java:1]: can you please merge the latest master into your branch if you've not yet done so?

this file shouldn't be generated in the latest master.

Comment (karzang) [reply samples/client/petstore/java/okhttp-gson-dynamicOperations/src/main/java/org/openapitools/client/model/BigCatAllOf.java]: i rebased on top of the master but the existing samples were not deleted when calling `./bin/generate-samples.sh`

I removed some samples that were falsely generated. Please let me know if you find more files which should not be generated.

**Pull request merged**